src/base/bittorrent/torrenthandle.cpp

```diff
@@ -1222,8 +1226,12 @@ void TorrentHandle::toggleSequentialDownload()
 
 void TorrentHandle::setFirstLastPiecePriority(bool b)
 {
-    if (!hasMetadata()) return;
+    if (!hasMetadata()) {
+        m_needSetFirstLastPiecePriority = b;
```

Comment:
Sequential download also needs to be saved with the resume data, so I'll fix that too.